requirements.txt

```diff
@@ -0,0 +1,3 @@
+pre-commit
+ruff
+black
```

Comment:
Wondering whether we should include `clang-format` here too, or should remove black and ruff here and just support them as pre-commit hooks (I believe pre-commit installs them by itself, no?)

Reply:
To avoid duplicate requirements in .pre-commit-config.yaml and requirements.txt, I feel like its best to just remove them from requirements.txt